pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -147,6 +155,72 @@ func GatherMostRecentMetrics(i *Gatherer) func() ([]record.Record, []error) {
 	}
 }
 
+// GatherContainerImages collects essential information about running containers.
+// Specifically, the age of pods, the set of running images and the container names are collected.
+//
+// Location in archive: config/running_containers.json
+func GatherContainerImages(i *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+		records := []record.Record{}
+
+		contInfo := ContainerInfo{
+			Images:     ContainerImageSet{},
+			Containers: PodsWithAge{},
+		}
+		// Cache for image indices in the collected set of container image.
+		image2idx := map[string]int{}
+
+		// Use the Limit and Continue fields to request the pod information in chunks.
+		continueValue := ""
+		for {
+			pods, err := i.coreClient.Pods("").List(metav1.ListOptions{
+				Limit:    imageGatherPodLimit,
+				Continue: continueValue,
+				// FieldSelector: "status.phase=Running",
+			})
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			for podIndex, pod := range pods.Items {
+				podPtr := &pods.Items[podIndex]
+				if strings.HasPrefix(pod.Namespace, "openshift") && hasContainerInCrashloop(podPtr) {
+					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pod.Namespace, pod.Name), Item: PodAnonymizer{podPtr}})
+				} else if pod.Status.Phase == corev1.PodRunning {
+					for _, container := range pod.Spec.Containers {
```

Comment:
This doesn't read init containers, so you're losing potentially half of the key values.  Or Ephemeral containers.